drivers/usb_c/tcpc/ucpd_numaker.c

```diff
@@ -0,0 +1,2555 @@
+	pwrctl |= UTCPD_PWRCTL_DSVBAM_DIS;
+	/* Disable UTCPD VBUS auto-discharge on disconnect
+	 * NOTE: UTCPD may not integrate with discharge, so this feature is
+	 *	 disabled and discharge is handled separately.
+	 */
+	pwrctl &= ~UTCPD_PWRCTL_ADGDC;
+	return NUMAKER_UTCPD_REG_WRITE_BY_NAME(dev, PWRCTL, pwrctl);
+}
+
+/**
+ * @brief Initializes UTCPD GPIO pins
+ *
+ * @retval 0 on success
+ * @retval -EIO on failure
+ */
+static int numaker_utcpd_gpios_init(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	int rc;
+	const struct gpio_dt_spec *spec;
+
+	/* Configure VBUS detect pin to INPUT to avoid intervening its power measurement */
+	spec = &config->utcpd.gpios.vbus_detect;
+	if (spec->port == NULL) {
+		LOG_ERR("VBUS detect pin not specified");
+		return -ENODEV;
+	}
+	if (!gpio_is_ready_dt(spec)) {
+		LOG_ERR("VBUS detect pin port device not ready");
+		return -ENODEV;
+	}
+	rc = gpio_pin_configure_dt(spec, GPIO_INPUT);
+	if (rc < 0) {
+		LOG_ERR("VBUS detect pin configured to INPUT failed: %d", rc);
+		return rc;
+	}
+
+	/* Configure VBUS discharge pin to OUTPUT INACTIVE */
+	spec = &config->utcpd.gpios.vbus_discharge;
+	if (spec->port != NULL) {
+		if (!gpio_is_ready_dt(spec)) {
+			LOG_ERR("VBUS discharge pin port device not ready");
+			return -ENODEV;
+		}
+		rc = gpio_pin_configure_dt(spec, GPIO_OUTPUT_INACTIVE);
+		if (rc < 0) {
+			LOG_ERR("VBUS discharge pin configured to OUTPUT INACTIVE failed: %d", rc);
+			return rc;
+		}
+	}
+
+	/* Configure VCONN discharge pin to OUTPUT INACTIVE */
+	spec = &config->utcpd.gpios.vconn_discharge;
+	if (spec->port != NULL) {
+		if (!gpio_is_ready_dt(spec)) {
+			LOG_ERR("VCONN discharge pin port device not ready");
+			return -ENODEV;
+		}
+		rc = gpio_pin_configure_dt(spec, GPIO_OUTPUT_INACTIVE);
+		if (rc < 0) {
+			LOG_ERR("VCONN discharge pin configured to OUTPUT INACTIVE failed: %d", rc);
+			return rc;
+		}
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Initializes UTCPD PHY
+ *
+ * @retval 0 on success
+ * @retval -EIO on failure
+ */
+static int numaker_utcpd_phy_init(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t phyctl = NUMAKER_UTCPD_REG_READ_BY_NAME(dev, PHYCTL);
+
+	/* Enable PHY
+	 *
+	 * NOTE: Only UTCPD0 is supported.
+	 */
+	SYS->UTCPDCTL |= SYS_UTCPDCTL_POREN0_Msk;
+	phyctl |= UTCPD_PHYCTL_PHYPWR_Msk;
+	return NUMAKER_UTCPD_REG_WRITE_BY_NAME(dev, PHYCTL, phyctl);
+}
+
+/**
+ * @brief Checks if UTCPD Dead Battery mode is enabled
+ *
+ * @retval true Dead Battery mode is enabled
+ * @retval false Dead Battery mode is not enabled
+ */
+static bool numaker_utcpd_deadbattery_query_enable(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t phyctl = NUMAKER_UTCPD_REG_READ_BY_NAME(dev, PHYCTL);
+
+	return (phyctl & UTCPD_PHYCTL_DBCTL_Msk) ? false : true;
```

Comment:
ok you can just do `return phyctl & UTCPD_PHYCTL_DBCTL_Msk;` here, since the return type is bool the compiler should do the rest for you

Reply:
ok, let compiler do the type conversion, but invert the return value to match  h/w register definition